### ISSUE TITLE
Automated Installation only works on 2nd attempt

### ISSUE THREAD

**Issue by Philipp4292** [Icinga/icinga-powershell-framework#535]
## Expected Behavior
We have a install script which automatically installs the framework and agent.
With version 1.9.0 it worked on the 1st run and the icinga2.exe and icinga-service.exe were running after the procedure.

## Current Behavior
On the first run, only the icinga-service.exe comes up.
Icinga2.exe cannot be installed because some cmdlets are not yet loaded.

The script output is:
```
[Notice]: Starting Icinga for Windows installation
[Notice]: Remote repository "Icinga Stable" was successfully added
[Notice]: Downloading "agent" from "http://BAR:42042/IfW/pub/icinga/IcingaForWindows/stable/agent/Icinga2-v2.13.3-x86_64.msi"
powershell.exe : The term 'Read-IcingaMSIMetadata' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:16358 char:20
+         $MSIData = & powershell.exe -Command {
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Read-IcingaMSIMetadata:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Installing component "agent" with version "" into "C:\Program Files\ICINGA2"
powershell.exe : The term 'Start-IcingaProcess' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again.
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:16378 char:27
+         $InstallProcess = & powershell.exe -Command {
+                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Start-IcingaProcess:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Error]: Failed to install component "agent":
False
[Notice]: Downloading "service" from "http://BAR:42042/IfW/pub/icinga/IcingaForWindows/stable/service/icinga-service-v1.2.0.zip"
[Notice]: Installing component "service" into "C:\Program Files\icinga-framework-service"
[Notice]: Updating Icinga PowerShell Service binary
[Notice]: Service User ".\LocalSystem" for service "icingapowershell" successfully updated
[Notice]: Stopping service "icingapowershell"
powershell.exe : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:3849 char:9
+         & powershell.exe -Command {
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Restarting service "icingapowershell"
powershell.exe : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:3600 char:9
+         & powershell.exe -Command {
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Stopping service "icingapowershell"
powershell.exe : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included,
verify that the path is correct and try again.
At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:3849 char:9
+         & powershell.exe -Command {
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Service User "NT Authority\NetworkService" for service "icingapowershell" successfully updated
[Notice]: Installation of component "service" was successful
[Notice]: Background daemon Cmdlet "Start-IcingaServiceCheckDaemon" has been configured
[Notice]: Downloading "plugins" from "http://BAR:42042/IfW/pub/icinga/IcingaForWindows/stable/plugins/icinga-powershell-plugins-1.9.0.zip"
[Notice]: Installing version "1.9.0" of component "plugins"
[Notice]: Unblocking Icinga PowerShell Files
[Notice]: Installation of component "plugins" with version "1.9.0" was successful. Open a new PowerShell to apply the changes
[Passed]: Icinga Agent is not installed and service is not present.
[Passed]: The specified user "NT Authority\NetworkService" is allowed to run as service
[Warning]: The specified directory "C:\ProgramData\icinga2\etc" was not found
[Warning]: The specified directory "C:\ProgramData\icinga2\var" was not found
[Failed]: Directory "C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache" is not accessible by the Icinga Service User "NT Authority\NetworkService"
\_ Please run the following command to fix this issue: Set-IcingaAcl -Directory 'C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache'
[Failed]: Directory "C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\config" is not accessible by the Icinga Service User "NT Authority\NetworkService"
\_ Please run the following command to fix this issue: Set-IcingaAcl -Directory 'C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\config'
[Warning]: The specified directory "C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\certificate" was not found
[Notice]: Stopping service "icingapowershell"
Write-IcingaConsoleError : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was
included, verify that the path is correct and try again.
At line:8 char:17
+                 Write-IcingaConsoleError -Message 'Failed to stop ser ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Restarting service "icingapowershell"
Write-IcingaConsoleError : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was
included, verify that the path is correct and try again.
At line:8 char:17
+                 Write-IcingaConsoleError -Message 'Failed to restart  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Background daemon Cmdlet "Start-IcingaWindowsRESTApi" has been configured
[Notice]: Stopping service "icingapowershell"
Write-IcingaConsoleError : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was
included, verify that the path is correct and try again.
At line:8 char:17
+                 Write-IcingaConsoleError -Message 'Failed to stop ser ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

[Notice]: Restarting service "icingapowershell"
Write-IcingaConsoleError : The term 'Write-IcingaConsoleError' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was
included, verify that the path is correct and try again.
At line:8 char:17
+                 Write-IcingaConsoleError -Message 'Failed to restart  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-IcingaConsoleError:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

## Possible Solution
Run script twice then the cmdlets are recognized.

## Steps to Reproduce (for bugs)
Start custom install script on Windows Server 2016 Datacenter. (changed hostname/ips and port)

```
[Net.ServicePointManager]::SecurityProtocol = 'tls12, tls11';
$ProgressPreference = 'SilentlyContinue';
[string]$ScriptFile = 'C:\Users\Public\IcingaForWindows.ps1';
Invoke-WebRequest `
-UseBasicParsing `
-Uri 'http://BAR:42042/IfW/pub/icinga/IcingaForWindows/IcingaForWindows.ps1' `
-OutFile $ScriptFile;
$InstallCommand=@'
{
 "IfW-ParentNodes":{"Values":["BAR"]},
 "IfW-Hostname":{"Selection":"5"},
 "IfW-Connection":{"Selection":"0"},
 "IfW-ParentZone":{"Values":["FOO"]},
 "IfW-ParentAddress":{"Values":{"BAR":["8.8.8.8"]}},
 "IfW-StableRepository":{"Values":["http://BAR:42042/pub/"]},
 "IfW-AgentVersion":{"Values":["2.13.3"]}
}
'@
& $ScriptFile `
-ModuleDirectory 'C:\Program Files\WindowsPowerShell\Modules\' `
-InstallCommand $InstallCommand `
-IcingaRepository "http://BAR:42042/pub/ifw.repo.json" `
-AllowUpdate;
```

## Context
Not affecting at all, because I can run the script twice.

## Your Environment
Major  Minor  Build  Revision
5      1      14393  2485

Version    BuildNumber Caption
10.0.14393 14393       Microsoft Windows Server 2016 Datacenter

**Comment by m4k5ym** [comment]
This is also for Windows Server 2012 R2 confirmed but in my case the whole powershell freezes as well.

**Comment by LordHepipud** [comment]
Thank you for the report. This is already fixed and will be released with v1.10.0: #522
